addons/website_slides/models/slide_channel.py

```diff
@@ -104,7 +159,9 @@ def _set_as_completed(self, completed=True):
             False if we remove user completion
         """
         for channel, memberships in self.grouped("channel_id").items():
-            memberships.completed = completed
+            for membership in memberships:
+                membership.member_status = 'completed' if completed else 'ongoing' if membership.completion > 0 else 'joined'
```

Comment:
I see this method is called in '_recompute_completion', seems a bit redundant ... Shouldn't the status be done in '_recompute_completion' and keep this one only for "one shot" karma update ? Somehow it seems some code cleaning (better relocating) would be welcomed to be sure each method has its own purpose.